### PR TITLE
Use Single Line Text field as Autocomplete Input

### PR DESCRIPTION
**Pull request by sbassah**
Original Snippets: https://github.com/gravitywiz/snippet-library/commit/3a3d93ce4642a305df434564195e45269c8e8159

### CODE REVIEW

**Comment by sbassah** [comment]
This is the PR for the conversion of the two snippets into one PHP snippet using the JS Snippet template.

File: gp-address-autocomplete/gpaa-single-line-as-input.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+ */
+class GPAA_Single_Line_Input {
+
+    public function __construct( $args = array() ) {
+
+        $this->_args = wp_parse_args( $args, array(
+            'form_id'  => false,
+            'address_field_id' => false,
+            'single_line_field_id'  => false,
+        ) );
```
Comment (spivurno) [inline gp-address-autocomplete/gpaa-single-line-as-input.php:23]: Code Standards: the equal signs should be aligned using spaces.

File: gp-address-autocomplete/gpaa-single-line-as-input.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+}
+
+// Configuration
+new GPAA_Single_Line_Input ( array(
+    'form_id'          => 123,          // The ID of your form.
+    'address_field_id'   => 4,         // The ID of the Address field.
+    'single_line_field_id' => 5        // The ID of the Single Line Text field..
+) );
```
Comment (spivurno) [inline gp-address-autocomplete/gpaa-single-line-as-input.php:109]: Code Standards: the equal signs should be aligned using spaces.

**Review by spivurno — APPROVED** [review]
Looks good! Approved with two minor tweaks.

File: gp-address-autocomplete/gpaa-single-line-as-input.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        $script = 'new GPAASingleLineInput( ' . json_encode( $args ) . ' );';
+        $slug   = implode( '_', array( 'gw_js_snippet_template', $this->_args['form_id'], $this->_args['address_field_id'] ) );
+
+        GFFormDisplay::add_init_script( $this->_args['form_id'], $slug, GFFormDisplay::ON_PAGE_RENDER, $script );
+
+    }
+
+    public function add_gpaa_init(){
+
+        $args['inputSelectors']['autocomplete'] = '#input_'.$this->_args['form_id'].'_'.$this->_args['single_line_field_id'];
```
Comment (spivurno) [inline gp-address-autocomplete/gpaa-single-line-as-input.php:89]: Code formatting: Add spaces before and after periods.

**Comment by sbassah** [comment]
Done with the code formatting.

**Comment by sbassah** [comment]
I guess we have to delete the two other snippets below from the Library and ask Dario to make a new usage video for this new snippet. 

[gpaa-single-line-as-input-value.js](https://github.com/gravitywiz/snippet-library/blob/master/gp-address-autocomplete/gpaa-single-line-as-input-value.js)
[gpaa-single-line-as-input-value.php](https://github.com/gravitywiz/snippet-library/blob/master/gp-address-autocomplete/gpaa-single-line-as-input.php)

**Comment by dariospace** [comment]
@spivurno no problema! I'll redo de video

**Pull request merged**